File: src/main/java/com/skyspace777/service/RewardRedemptionService.java
package com.skyspace777.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspace777.domain.RewardRedemption;
import com.skyspace777.dto.RewardRedemptionDTO;
import com.skyspace777.dto.RewardRedemptionSearchDTO;
import com.skyspace777.dto.RewardRedemptionPageDTO;
import com.skyspace777.dto.RewardRedemptionConvertCriteriaDTO;
import com.skyspace777.service.GenericService;
import com.skyspace777.dto.common.RequestDTO;
import com.skyspace777.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface RewardRedemptionService extends GenericService<RewardRedemption, Integer> {

	List<RewardRedemption> findAll();

	ResultDTO addRewardRedemption(RewardRedemptionDTO rewardRedemptionDTO, RequestDTO requestDTO);

	ResultDTO updateRewardRedemption(RewardRedemptionDTO rewardRedemptionDTO, RequestDTO requestDTO);

    Page<RewardRedemption> getAllRewardRedemptions(Pageable pageable);

    Page<RewardRedemption> getAllRewardRedemptions(Specification<RewardRedemption> spec, Pageable pageable);

	ResponseEntity<RewardRedemptionPageDTO> getRewardRedemptions(RewardRedemptionSearchDTO rewardRedemptionSearchDTO);
	
	List<RewardRedemptionDTO> convertRewardRedemptionsToRewardRedemptionDTOs(List<RewardRedemption> rewardRedemptions, RewardRedemptionConvertCriteriaDTO convertCriteria);

	RewardRedemptionDTO getRewardRedemptionDTOById(Integer rewardRedemptionId);







}

File: skyspace777_ui/src/services/RewardRedemptionService.js
import http from "../http-common"; 

class RewardRedemptionService {
  getAllRewardRedemptions(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/rewardRedemption/rewardRedemptions`, searchDTO);
  }

  get(rewardRedemptionId) {
    return this.getRequest(`/rewardRedemption/${rewardRedemptionId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/rewardRedemption?field=${matchData}`, null);
  }

  addRewardRedemption(data) {
    return http.post("/rewardRedemption/addRewardRedemption", data);
  }

  update(data) {
  	return http.post("/rewardRedemption/updateRewardRedemption", data);
  }
  
  uploadImage(data,rewardRedemptionId) {
  	return http.postForm("/rewardRedemption/uploadImage/"+rewardRedemptionId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new RewardRedemptionService();
